### PR TITLE
fix: cmdline fallback mapping for <S-Tab>

### LINKED ISSUE #953: Incorrect cmdline default mapping for `<S-Tab>`
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/15f08a8faa22d52480cdcb9ef9ca698120f04363/doc/cmp.txt#L616) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/hrsh7th/nvim-cmp/issues) and there are no open or closed issues with the same problem.

### Neovim Version

0.7

### Minimal reproducible full config

```vim
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-buffer'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
call plug#end()
PlugInstall | quit

" Setup global configuration. More on configuration below.
lua << EOF
local cmp = require "cmp"
cmp.setup {
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },

  mapping = {
    ['<CR>'] = cmp.mapping.confirm({ select = true })
  },

  sources = {
    { name = "nvim_lsp" },
    { name = "buffer" },
  },
}

cmp.setup.cmdline(':', {
  mapping = cmp.mapping.preset.cmdline(),
  sources = cmp.config.sources({
    { name = 'path' }
  }, {
    { name = 'cmdline' }
  })
})

EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').update_capabilities(vim.lsp.protocol.make_client_capabilities())

require'lspconfig'.cssls.setup {
  capabilities = capabilities,
}
EOF

```

### Description

Pressing `<S-Tab>` performs the same action as `<Tab>`, even though separate functions are called. This is due to https://github.com/hrsh7th/nvim-cmp/blob/baa8646c248486fdd46e8a087ae5623165901b9f/lua/cmp/config/mapping.lua#L65

### Steps to reproduce

`:P<tab>` brings up the default wildmenu (not configured with cmp). Pressing tab again will cycle to next entries.

### Expected behavior

Pressing shift tab should move you to previous entries.

### Actual behavior

Pressing shift tab cycles you to next entries (same as Tab)

### Additional context

Also, just noticed this from the README, but is there a difference between the `cmp.config.sources` call in https://github.com/hrsh7th/nvim-cmp/blob/baa8646c248486fdd46e8a087ae5623165901b9f/README.md#L112 and https://github.com/hrsh7th/nvim-cmp/blob/baa8646c248486fdd46e8a087ae5623165901b9f/README.md#L120?

### PR DESCRIPTION
**Pull request by axieax**
Closes #953

### CODE REVIEW

**Comment by hrsh7th** [comment]
In my environment, this fix breaks `<S-Tab>` mapping.

I think the following keys is valid... WDYT?

```lua
    ['<Tab>'] = {
      c = function()
        local cmp = require('cmp')
        if cmp.visible() then
          cmp.select_next_item()
        elseif vim.fn.pumvisible() == 1 then
          feedkeys.call(keymap.t('<Tab>'), 'n')
        else
          feedkeys.call(keymap.t('<C-z><C-n>'), 'n')
        end
      end,
    },
    ['<S-Tab>'] = {
      c = function()
        local cmp = require('cmp')
        if cmp.visible() then
          cmp.select_prev_item()
        elseif vim.fn.pumvisible() == 1 then
          feedkeys.call(keymap.t('<S-Tab>'), 'n')
        else
          feedkeys.call(keymap.t('<C-z><C-p>'), 'n')
        end
      end,
    },
```

**Comment by axieax** [comment]
Hmm in my environment `<Tab>` skips an entry (moves twice instead of once), and `<S-Tab>` still doesn't work for some reason

https://user-images.githubusercontent.com/62098008/166261764-6ce5fa86-d135-4495-8942-480e10e36146.mp4

.

File: lua/cmp/config/mapping.lua [hunk 1]
```diff
@@ -62,7 +62,7 @@ mapping.preset.cmdline = function(override)
         if cmp.visible() then
           cmp.select_prev_item()
         else
-          feedkeys.call(keymap.t('<C-z>'), 'n')
+          feedkeys.call(keymap.t('<S-Tab>'), 'n')
```
Comment (eyalk11) [inline lua/cmp/config/mapping.lua:65]: feedkeys is problematic as sometimes people change the default. `normal!` is the way.

Comment (axieax) [reply lua/cmp/config/mapping.lua:65]: @eyalk11 do you have an example for how would that be used here?

**Pull request closed without merging**